crates/wit-parser/src/decoding.rs

```diff
@@ -20,6 +26,22 @@ struct ComponentInfo {
     types: types::Types,
     /// List of all imports and exports from this component.
     externs: Vec<(String, Extern)>,
+    /// Packages
+    explicit: Vec<ExplicitPackageInfo>,
+    /// Decoded package metadata
+    package_metadata: Option<PackageMetadata>,
+}
+
+/// Represents information about a decoded WebAssembly component.
+#[derive(Default)]
+struct ExplicitPackageInfo {
```

Comment:
I'd prefer to avoid the duplication between `ComponentInfo` and `ExplicitPackageInfo` if possible. Could there perhaps be a single `PackageInfo` structure which has `externs` and a list of nested info? For example `externs` would track imports and exports and `explicit` (perhaps modulo a renaming) would store nested components. The parse loop would then recurse for nested components rather than handling everything in a single loop with an if/else on the depth.